main.tf

```diff
@@ -0,0 +1,214 @@
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = aws_s3_bucket.athena_results[0].id
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_object" "sales_val_flagging" {
+  bucket = local.s3_bucket_glue_assets
+  key    = "${local.s3_prefix}/sales_val_flagging.py"
+  source = "${path.module}/glue/sales_val_flagging.py"
+  etag   = filemd5("${path.module}/glue/sales_val_flagging.py")
+}
+
+resource "aws_s3_object" "flagging_script" {
+  bucket = local.s3_bucket_glue_assets
+  key    = "${local.s3_prefix}/flagging.py"
+  source = "${path.module}/glue/flagging_script_glue/flagging.py"
+  etag   = filemd5("${path.module}/glue/flagging_script_glue/flagging.py")
+}
+
+resource "aws_s3_object" "sale_flag_metadata" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = local.s3_bucket_data_warehouse
+  key    = "sale/flag/sale_flag_schema.parquet"
+  source = "${path.module}/glue/schema/sale_flag_schema.parquet"
+  etag   = filemd5("${path.module}/glue/schema/sale_flag_schema.parquet")
+}
+
+resource "aws_glue_job" "sales_val_flagging" {
+  name            = local.glue_job_name
+  role_arn        = var.iam_role_arn
+  max_retries     = 0
+  max_capacity    = "1.0"
+  glue_version    = "3.0"
+  execution_class = "STANDARD"
+
+  command {
+    name            = "pythonshell"
+    script_location = "s3://${aws_s3_object.sales_val_flagging.bucket}/${aws_s3_object.sales_val_flagging.key}"
+    python_version  = "3.9"
+  }
+
+  default_arguments = {
+    "--s3_glue_bucket"            = local.s3_bucket_glue_assets
+    "--s3_prefix"                 = "${local.s3_prefix}/"
+    "--aws_s3_warehouse_bucket"   = "s3://${local.s3_bucket_data_warehouse}"
+    "--enable-job-insights"       = false
+    "--region_name"               = "us-east-1"
+    "--job-language"              = "python"
+    "--TempDir"                   = "s3://${local.s3_bucket_glue_assets}/temporary/"
+    "--s3_staging_dir"            = "s3://ccao-athena-results-us-east-1"
+    "--stat_groups"               = "rolling_window,township_code,class"
+    "--iso_forest"                = "meta_sale_price,sv_price_per_sqft,sv_days_since_last_transaction,sv_cgdr,sv_sale_dup_counts"
+    "--rolling_window_num"        = 12
+    "--time_frame_start"          = "2023-01-01"
+    "--dev_bounds"                = "2,3"
+    "--additional-python-modules" = "Random-Word==1.0.11,boto3==1.28.12,pandas==2.0.2"
+    "--commit_sha"                = var.commit_sha
+    "--min_groups_threshold"      = "30"
+    "--ptax_sd"                   = "1,1"
+    "--sale_flag_table"           = "${local.athena_database_name}.flag"
+  }
+}
+
+resource "aws_athena_database" "sale_test" {
+  count         = terraform.workspace == "prod" ? 0 : 1
+  name          = local.athena_database_name
+  comment       = "Test sale database for the ${terraform.workspace} branch"
+  bucket        = aws_s3_bucket.athena_results[0].id
+  force_destroy = true
+}
```

Comment:
New as of last review cycle, this staging database lets us keep our testing data isolated from production data.

Reply:
**suggestion:** You're gonna hate me for this, but can we prefix the ephemeral buckets with `ci-` or `dev-`, as we did for the Athena resources? It keeps them separately sorted in the UI and also helps differentiate their purpose.

Reply:
No problem, done in efd94bab!